### PR TITLE
Replaces About screen with unified Automattic About screen

### PR DESCRIPTION
**Pull request by frosty**
### Description

This PR introduces the new [unified Automattic About screen](https://github.com/Automattic/AutomatticAbout-Swift) to the app, replacing the existing About screen. This contains links to social accounts, websites, the ability to rate and share the app, and links to legal policies and third party licenses.

I noticed that the original About screen included a crash debugging gesture (tapping 4 times on the app icon). The new About screen framework currently doesn't support adding functionality like this directly to the header view, so for now I've moved this debugging gesture to the heart icon in the bottom of the Settings screen. If this isn't acceptable, I can update the AutomatticAbout framework to allow a custom gesture to be added to the header view.

The other known issue is that I ran out of time to add analytics events to the various rows within the About screen. However, I don't think the previous screen had this either, so this isn't a regression and can be added as a future enhancement.
 
### Testing instructions

* Build and run the app and navigate to Menu > Settings > WooCommerce (under 'About the App').
* Interact with each item in the screen and ensure it performs as you'd expect. The "Share with a Friend" item may be easier to test on a device, as you're very limited for sharing options on the Simulator.
* On the Settings screen, try tapping the heart icon 4 times quickly to test the crash debug gesture.

### Screenshots

|   |   |
|---|---|
| Simulator Screen Shot - iPhone 13 Pro - 2022-03-11 at 22 43 14 | Simulator Screen Shot - iPhone 13 Pro - 2022-03-11 at 22 52 42 |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Tools/AppRatings/AppRatingManager.swift
Comment (astralbodies) [inline WooCommerce/Classes/Tools/AppRatings/AppRatingManager.swift:309]: Do you know if this URL works for all of the different App Stores? I don't have an account in the UK store to test. 😄

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (astralbodies) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: Can you override this value and call the app WooCommerce? The shortened name of "Woo" feels weird when I see it on the screen. 😉

Comment (frosty) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: Sure!

File: WooCommerce/Classes/Tools/AppRatings/AppRatingManager.swift
Comment (frosty) [reply WooCommerce/Classes/Tools/AppRatings/AppRatingManager.swift:309]: I tested it in the UK and it works fine for me 🙂

**Comment by frosty** [comment]
@astralbodies Thanks for the review! I've updated the app name as requested:

File: WooCommerce/Classes/Tools/ShareAppTextItemActivitySource.swift
Comment (astralbodies) [inline WooCommerce/Classes/Tools/ShareAppTextItemActivitySource.swift]: TIL about Self with the capital S!

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (astralbodies) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]:
```suggestion
        static let share             = NSLocalizedString("Share with Friends", comment: "Title for button allowing users to share "
```

**Review by astralbodies — APPROVED** [review]
Thanks for doing this, @frosty!

**Comment by astralbodies** [comment]
I tested on my iPhone 12 Pro Max with the installable build. Everything worked as expected!

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift]: nit: maybe weak self just in case?
```suggestion
        present(controller, animated: true) { [weak self] in
            self?.tableView.deselectSelectedRowWithAnimation(true)
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift]: nit: maybe this string can be in the `Localization` enum to be consistent with other string usages in this class? (similar for other 2 strings below)

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift]: nit: I think we can mark this extension as `private` and remove it from the functions below, it works for `@objc` function too
```suggestion
private extension SettingsViewController {
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: nit: can this class be final?
```suggestion
final class WooAboutScreenConfiguration: AboutScreenConfiguration {
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: nit: we usually put all localizable strings in a `Localization` enum based on the [documentation](https://github.com/woocommerce/woocommerce-ios/blob/trunk/docs/naming-conventions.md#string-constants-in-nested-enums) 🙂

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: nit: does it make sense to localize these two links for other languages/locales? like in Spanish, the blog is at `https://woocommerce.com/es-es/blog/`

File: WooCommerce/Classes/Tools/ShareAppTextItemActivitySource.swift
Comment (jaclync) [inline WooCommerce/Classes/Tools/ShareAppTextItemActivitySource.swift]: nit: could this be private? usually, we keep the strings in a `Localization` enum based on the [doc](https://github.com/woocommerce/woocommerce-ios/blob/trunk/docs/naming-conventions.md#string-constants-in-nested-enums)

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: nit: usually we keep private functions in a separate private extension so it's easier to tell what's public and necessary to conform to a protocol (`AboutScreenConfiguration` in this case) from the private helpers 😄

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift]: nice debugging tool!

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift:19]: TIL Woo has an Instagram account 😆

**Review by jaclync — APPROVED** [review]
Nice work @frosty, thanks for adding this! Love the new About screen, especially the Automattic family icons! Would be really cool if I can click on each app icon that links to the App Store 😁

- One minor thing that used to be in the old About screen is the full app version like `8.7.0.2` while the new screen just shows `8.7`. I checked WPiOS in production and it actually shows both, is this something we can do in Woo too?

old About screen | new About screen | WPiOS
-- | -- | --
Simulator Screen Shot - iPhone 11 - 2022-03-14 at 10 10 38 | Simulator Screen Shot - iPhone 11 - 2022-03-14 at 11 01 25 | IMG_0114
- Nice-to-have: WPiOS shows the Twitter handle in the Twitter row, maybe we can do the same for Twitter and/or Instagram rows in Woo 😃

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (frosty) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: I just tested this by setting my region for Safari to Spain, and I was automatically redirected to the Spanish blog from `woocommerce.com/blog`. So I think it should be fine to leave as-is and it'll happen automatically on the server side. Thanks for raising the question though!

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (frosty) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: Thanks for the tip! I hadn't seen this, and have now updated this across my changes.

**Comment by frosty** [comment]
@jaclync Thanks so much for the helpful comments! I've improved the structure of my changes, in particular with where the localised strings are defined. Ready for another look!

**Comment by astralbodies** [comment]
Awesome review, @jaclync - I need to do PR reviews more to keep my chops up 😸

**Comment by Ecarrion** [comment]
Hi @frosty thanks for doing this! 🙇 

A little question (which does not block this PR at all) is to know if there is a way to change the title font? The current one does not look in line with our current titles, which I think is `SF Pro`

Screen Shot 2022-03-14 at 10 53 49 AM

**Comment by frosty** [comment]
@Ecarrion Yep, that's supported by the framework – thanks for asking! How does this look?

Simulator Screen Shot - iPhone 13 Pro - 2022-03-14 at 20 18 37

**Comment by Ecarrion** [comment]
@frosty That looks great! 🚀

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: nit: could this be `final`?
```suggestion
final class WooLegalAndMoreSubmenuConfiguration: AboutScreenConfiguration {
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: super nit: 
```suggestion
    enum Links {
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: super nit:
```suggestion
    static func iconNameFromBundle() -> String {
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: super nit:
```suggestion
    enum Links {
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/About/WooAboutScreenConfiguration.swift]: Tha's nice the locale/language is automatically handled!

**Review by jaclync — APPROVED** [review]
Thanks for the updates @frosty, look great! For the font update https://github.com/woocommerce/woocommerce-ios/pull/6421#issuecomment-1067249068, have you pushed the change? I still see the WP style font 🤔

**Comment by frosty** [comment]
@jaclync I hadn't pushed the changes for the font as I just wanted confirmation that it looked okay before doing so 🙂  Have pushed them up now though!

**Review by jaclync — APPROVED** [review]
The header font looks great! Still LGTM :shipit: (the CI failures look unrelated to this PR, should be good after merging `trunk` to the PR branch again)

**Comment by frosty** [comment]
Thanks so much for the reviews, @jaclync!

**Pull request merged**
